### PR TITLE
feat: add files api

### PR DESCRIPTION
**Pull request by skjnldsv**
See 
- https://github.com/nextcloud/documentation/pull/10414#discussion_r1200228653
- https://github.com/nextcloud/server/pull/37824

### CODE REVIEW

File: developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_27.rst [hunk 1]
```diff
@@ -25,13 +25,7 @@ Front-end changes
...
 ^^^^^^^^^^
 
-* A new Files Router API allows you to control the files router service and update views, querys or param without page reload (`nextcloud/server#37824 <https://github.com/nextcloud/server/pull/37824>`_)
-
-  .. code-block:: js
-
-    // https://github.com/nextcloud/server/blob/master/apps/files/src/services/RouterService.ts
-    OCP.Files.Router.goTo('/trashbin?dir=/Unsplash.d1680193199')
-    OCP.Files.Router.goToRoute('fileslist', { view: 'files' }, { dir: '/Folders/Group folder' })
+* A new Files Router API allows you to control the files router service and update views, querys or param without page reload. See :doc:`../../client_apis/files`
```
Comment (ChristophWurst) [inline developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_27.rst:28]: add a label and use :ref: so the link doesn't break when we move pages around https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#use-a-custom-anchor

File: developer_manual/client_apis/files.rst [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+Router
+^^^^^^
+
+To change views and parameters of the current Files app, we expose the router.
+This directly maps to the Vue router. You can check their own `documentation <https://router.vuejs.org/guide/essentials/navigation.html#navigate-to-a-different-location>`_ to
+better understand the methods.
+
+.. code-block:: ts
```
Comment (ChristophWurst) [inline developer_manual/client_apis/files.rst:24]: does this render as typescript?

Comment (skjnldsv) [reply developer_manual/client_apis/files.rst:24]: Seems to yeah

Comment (nickvergessen) [reply developer_manual/client_apis/files.rst:27]: PR template asks for a screenshot which would have been helpful to see whether it does:
grafik

File: developer_manual/client_apis/index.rst [hunk 3]
```diff
@@ -7,11 +7,12 @@ Clients and Client APIs
 .. toctree::
     :maxdepth: 2
 
-    general
```
Comment (nickvergessen) [inline developer_manual/client_apis/index.rst:10]: This was sorted by priority and I think having general first would still be helpful, although I get the alphabetic one.
So maybe alphabetic with the exception of general?

Comment (skjnldsv) [reply developer_manual/client_apis/index.rst:10]: Yes :+1:

Comment (skjnldsv) [reply developer_manual/client_apis/index.rst:10]: Done

File: developer_manual/client_apis/files.rst [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+.. sectionauthor:: John Molakvoæ <skjnldsv@protonmail.com>
+
+Since Nextcloud 27, the files app has been rewritten to use new standards
+and frameworks. This means that the old documentation is no longer valid.
+We will update and document the new files app APIs and methods here.
+
+.. note:: Some external libraries offers in-depth technical documentation.
+    Please have a look at the following:
+
+    * https://github.com/nextcloud/nextcloud-files (`documentation <https://nextcloud.github.io/nextcloud-files/>`_)
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply developer_manual/client_apis/files.rst]: Done

File: developer_manual/client_apis/files.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
+.. _FilesAPI:
+
+=====
+Files
+=====
+
+.. sectionauthor:: John Molakvoæ <skjnldsv@protonmail.com>
+
+Since Nextcloud 27, the files app has been rewritten to use new standards
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply developer_manual/client_apis/files.rst:10]: Done

**Pull request merged**